src/environs/__init__.py

```diff
@@ -569,6 +569,27 @@ def _expand_vars(self, parsed_key, value):
     def _get_key(self, key: _StrType, *, omit_prefix: _BoolType = False) -> _StrType:
         return self._prefix + key if self._prefix and not omit_prefix else key
 
+    def _get_value(self, env_key: _StrType, default: typing.Any) -> typing.Any:
+        return os.environ.get(env_key, default)
+
+
+class FileAwareEnv(Env):
+    """An environment variable reader that supports reading values from files."""
+
+    def __init__(self, *, file_suffix: _StrType = "_FILE", **kwargs):
+        self.file_suffix = file_suffix
+        super().__init__(**kwargs)
+
+    def _get_value(self, env_key: _StrType, default: typing.Any) -> typing.Any:
+        """Return the contents of the file referenced in key <env_key>_FILE, if present."""
+        file_key = f"{env_key}{self.file_suffix}"
+        if file_path := os.environ.get(file_key, None):
+            try:
+                return Path(file_path).read_text()
+            except (FileNotFoundError, IsADirectoryError, PermissionError) as err:
+                raise ValueError("path should exist and be a readable file") from err
```

Comment:
would be useful to have the relevant envvar in the error message. something like

```suggestion
                raise ValueError(f"The value of {file_key} must be a readable file path") from err
```